### PR TITLE
[FLINK-17424][e2e] Adopt download strategy of Kafka tests in Elasticsearch tests

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

Adds some retry and timeout logic to the Elasticsearch binary download logic.

Additionally, it cleans up the testing matrix a bit to make the e2e tests quicker on Azure.

## Brief change log

See commit messages.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-end-to-end-tests/test-scripts/test_sql_client.sh [hunk 1]
```diff
@@ -112,9 +111,7 @@ function prepare_elasticsearch {
       *)          OS_TYPE="UNKNOWN:${unameOut}"
   esac
 
-  if [[ "$ELASTICSEARCH_VERSION" == 6 ]]; then
-    DOWNLOAD_URL=$ELASTICSEARCH6_DOWNLOAD_URL
-  elif [[ "$ELASTICSEARCH_VERSION" == 7 ]] && [[ "$OS_TYPE" == "mac" ]]; then
+  if [[ "$ELASTICSEARCH_VERSION" == 7 ]] && [[ "$OS_TYPE" == "mac" ]]; then
```
Comment (leonardBang) [inline flink-end-to-end-tests/test-scripts/test_sql_client.sh:114]: minor: after deleted the test for Es6, the version check can be ignored

**Review by leonardBang — COMMENTED** [review]
Thanks @twalthr for the contribution, I left only one minor comment

**Comment by twalthr** [comment]
Thanks @leonardBang. I removed the version as a parameter. I will merge this if there are no objections.

**Review by leonardBang — APPROVED** [review]
Thanks for the update @twalthr ,  +1 to merge once the build pass

**Pull request closed without merging**